### PR TITLE
[2.5][CVE-2022-37601][CVE-2022-37599] Bump loader-utils to 2.0.4

### PR DESCRIPTION
**Pull request by ananzh**
Issue Resolved:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3306

Signed-off-by: Anan Zhuang <ananzh@amazon.com>

 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ananzh** [comment]
tests fail due to no 2.5.1 snapshots
but 2.x tests are all good

File: yarn.lock [hunk 1]
```diff
@@ -11807,10 +11807,10 @@ loader-runner@^2.4.0:
   resolved "https://registry.yarnpkg.com/loader-runner/-/loader-runner-2.4.0.tgz#ed47066bfe534d7e84c4c7b9998c2a75607d9357"
   integrity sha512-Jsmr89RcXGIwivFY21FcRrisYZfvLMTWx5kOLc+JTxtpBOG6xML0vzbc6SEQG2FO9/4Fc3wW4LVcB5DmGflaRw==
 
-loader-utils@^1.2.3, loader-utils@^2.0.0, loader-utils@^2.0.3:
-  version "2.0.3"
-  resolved "https://registry.yarnpkg.com/loader-utils/-/loader-utils-2.0.3.tgz#d4b15b8504c63d1fc3f2ade52d41bc8459d6ede1"
-  integrity sha512-THWqIsn8QRnvLl0shHYVBN9syumU8pYWEHPTmkiVGd+7K5eFNVSY6AJhRvgGF70gg1Dz+l/k8WicvFCxdEs60A==
+loader-utils@^1.2.3, loader-utils@^2.0.0, loader-utils@^2.0.4:
```
Comment (joshuarrrr) [inline yarn.lock:11810]: Where is the `loader-utils@^1.2.3` still coming from? Given the updates to `osd-shared-deps` and `osd-optimizer`, I wouldn't think we'd still need a resolution from that version. From a nested dep?

**Review by joshuarrrr — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (ananzh) [reply yarn.lock:11810]: Yeah 1.2.3 is from 

```
string-replace-loader@^2.2.0:
  version "2.3.0"
  resolved "https://registry.yarnpkg.com/string-replace-loader/-/string-replace-loader-2.3.0.tgz#7f29be7d73c94dd92eccd5c5a15651181d7ecd3d"
  integrity sha512-HYBIHStViMKLZC/Lehxy42OuwsBaPzX/LjcF5mkJlE2SnHXmW6SW6eiHABTXnY8ZCm/REbdJ8qnA0ptmIzN0Ng==
  dependencies:
    loader-utils "^1.2.3"
    schema-utils "^2.6.5"
```

**Comment by ananzh** [comment]
Test failed due to `Snapshots for 2.5.1 are not available`. Below is an example.

```
● uiSettings/routes › doc exists › delete route › returns a 200 and deletes the setting

    Snapshots for 2.5.1 are not available

      31 |  */
      32 | exports.createCliError = function (message) {
    > 33 |   const error = new Error(message);
         |                 ^
      34 |   error.isCliError = true;
      35 |   return error;
      36 | };

      at createCliError (packages/osd-opensearch/target/errors.js:33:17)
      at getArtifactSpecForSnapshotFromUrl (packages/osd-opensearch/target/artifact.js:262:11)
          at runMicrotasks (<anonymous>)
      at Function.getSnapshot (packages/osd-opensearch/target/artifact.js:299:26)
      at Object.installSnapshot [as downloadSnapshot] (packages/osd-opensearch/target/install/snapshot.js:73:20)
      at installSnapshot (packages/osd-opensearch/target/install/snapshot.js:105:7)
      at Cluster.installSnapshot (packages/osd-opensearch/target/cluster.js:179:9)
      at OpenSearchTestCluster.start (packages/osd-test/src/legacy_opensearch/legacy_opensearch_test_cluster.js:95:24)
      at Object.startOpenSearch (src/core/test_helpers/osd_server.ts:265:7)
      at Object.startServers (src/core/server/ui_settings/integration_tests/lib/servers.ts:70:22)
```

**Comment by ananzh** [comment]
2.5 is not an active branch any more since we cut 2.6. close it

**Pull request closed without merging**